rust/libnewsboat/src/utils.rs

```diff
@@ -620,6 +620,25 @@ pub fn strip_comments(line: &str) -> &str {
     &line[0..first_pound_chr_idx]
 }
 
+/// Extract filter and url from line separated by ':'.
+pub fn extract_filter(line: &str) -> (&str, &str) {
+    debug_assert!(line.starts_with("filter:"));
+    // line must start with "filter:"
+    let line = line.get("filter:".len()..).unwrap();
+    // should contain url separated by ':' but no guarantee
+    let (filter, url) = line.split_at(line.find(':').expect("should contain url"));
```

Comment:
This `expect` will cause panic if `line` is something like `filter:text-without-colon`, right? How did original C++ implementation behave?

Reply:
```c++
        std::string::size_type pos = line.find_first_of(":", 0);
        std::string::size_type pos1 = line.find_first_of(":", pos + 1);
        filter = line.substr(pos + 1, pos1 - pos - 1);
        pos = pos1;
        url = line.substr(pos + 1, line.length() - pos);
        LOG(Level::DEBUG,
                "utils::extract_filter: %s -> filter: %s url: %s",
                line,
                filter,
                url);
```
By looking at the original C++ code, it seemed like it assumed that it have `:`, just by quick scanning it does not check.

Reply:
Yeah, I just tested on `master`, and in that case it returns the input string. Doesn't crash the application, though, so please do something about that `expect`.

Reply:
Now it should accept the input even if it does not contain the second `:`.